sse2neon.h

```diff
@@ -2501,31 +2511,152 @@ FORCE_INLINE __m128 _mm_rsqrt_ss(__m128 in)
+    const uint32x4_t vexp_mask = {0x7f800000, 0x7f800000, 0x7f800000,
+                                  0x7f800000};
+    const uint32x4_t vqnan_bit_mask = {0x00400000, 0x00400000, 0x00400000,
+                                       0x00400000};
+    const uint32x4_t vsnan_bit_mask = {0x003fffff, 0x003fffff, 0x003fffff,
+                                       0x003fffff};
+    const uint32x4_t vzero = vdupq_n_u32(0);
+
+    uint32x4_t v_exp_all_ones =
+        vceqq_u32(vandq_u32(vreinterpretq_u32_f32(a), vexp_mask), vexp_mask);
+
+    // Check if qnan mantissa bits is off
+    uint32x4_t v_qnan_bit_off =
+        vceqq_u32(vandq_u32(vreinterpretq_u32_f32(a), vqnan_bit_mask), vzero);
+    uint32x4_t v_snan_bit_any =
+        vcgtq_u32(vandq_u32(vreinterpretq_u32_f32(a), vsnan_bit_mask), vzero);
+
+    uint32x4_t v_is_snan =
+        vandq_u32(vandq_u32(v_exp_all_ones, v_qnan_bit_off), v_snan_bit_any);
+
+    return v_is_snan;
+}
+
+// Check if input is NaN (sNaN or qNan)
+FORCE_INLINE uint32x4_t _sse2neon_is_nan(float32x4_t a)
+{
+    const uint32x4_t vexp_mask = {0x7f800000, 0x7f800000, 0x7f800000,
+                                  0x7f800000};
+    const uint32x4_t vmantissa_mask = {0x007fffff, 0x007fffff, 0x007fffff,
+                                       0x007fffff};
+    const uint32x4_t vzero = vdupq_n_u32(0);
+
+    // Check if all exp bits are 1.
+    uint32x4_t v_exp_all_ones =
+        vceqq_u32(vandq_u32(vreinterpretq_u32_f32(a), vexp_mask), vexp_mask);
+
+    // Check if any mantissa bits are on(qNaN or sNaN)
+    uint32x4_t v_mantissa_any =
+        vcgtq_u32(vandq_u32(vreinterpretq_u32_f32(a), vmantissa_mask), vzero);
+
+    return vandq_u32(v_exp_all_ones, v_mantissa_any);
+}
+
+// Accurate simulation of _mm_min_ps using NEON
+FORCE_INLINE float32x4_t _sse2neon_vmin(float32x4_t a, float32x4_t b)
+{
+    // when both input are (+/-)0.0, return the second
+    // when the first input is NaN (sNaN or qNaN), return the second.
+    // when the second input is sNaN, return sNaN (return the second).
+    // otherwise return min(a, b)
+    const float32x4_t vzero = vdupq_n_f32(0.0f);
+    const uint32x4_t v_src1_is_snan = _sse2neon_is_snan(b);
+
+    // fortunately, ceqq_f32 ignores the sign.
+    const uint32x4_t v_both_are_zeros =
+        vandq_u32(vceqq_f32(a, vzero), vceqq_f32(b, vzero));
+
+    const uint32x4_t v_src0_is_nan = _sse2neon_is_nan(a);
+
+    const float32x4_t v_min = vminq_f32(a, b);
+
+    float32x4_t v_special_case = vbslq_f32(v_both_are_zeros, b, v_min);
+    v_special_case = vbslq_f32(v_src0_is_nan, b, v_special_case);
+    v_special_case = vbslq_f32(v_src1_is_snan, b, v_special_case);
+
+    // Requie NaN or both zero case handling?
+    const uint32x4_t v_require_special_handling =
+        vorrq_u32(v_src1_is_snan, vorrq_u32(v_both_are_zeros, v_src0_is_nan));
+
+    // use min(a, b) when !(require special handling)
+    return vbslq_f32(v_require_special_handling, v_special_case, v_min);
```

Comment:
Isn't `v_special_case` the required return value?

Reply:
@syoyo, Could you check the special case?

Reply:
@jserv I can't understand what @marktwtn is saying. BTW, you can omit src1 snan check. I found checking src1 snan  is redundant after I wrote the initial code.

Reply:
I mean it seems like variable `v_special_case` and `vbslq_f32(v_require_special_handling, v_special_case, v_min)` have the same value?
If it is, then the line 2583~2589 should be removed.

Reply:
@syoyo any idea?

Reply:
@marktwtn as I said you are better to write unit tests.

Reply:
@jserv please remove the redundant code and return the `v_special_case`.